python/cudf/dataframe/series.py

```diff
@@ -460,46 +461,255 @@ def _unaryop(self, fn):
         outcol = self._column.unary_operator(fn)
         return self._copy_construct(data=outcol)
 
+    def _filled_binaryop(self, other, fn, fill_value=None, reflect=False):
+        def func(lhs, rhs):
+            return fn(rhs, lhs) if reflect else fn(lhs, rhs)
+
+        if fill_value is not None:
+            if isinstance(other, Series):
+                if self.has_null_mask and other.has_null_mask:
```

Comment:
Shouldn't we only go down this path if `fill_value` is `None`?

Reply:
Actually no, the default behaviour for any binary op is to simply AND the masks of both the operands. However this is slightly different when a `fill_value` is specified. In this case, the result of an operation is like this:

|lhs|rhs|result|
|-|-|-|
|`x`|`Null`|`op(x, fill_value)`|
|`Null`|`y`|`op(fill_value, y)`|
|`Null`|`Null`|`Null`|

Case 3 is where the trouble is. This table can basically be achieved by doing an OR of the two masks. The extra check added to see if both the series even have a null mask will ensure we don't get an error in either of these
```python
lmask = Series(data=self.nullmask)
rmask = Series(data=other.nullmask)
```
If one of the series does not have a null mask at all then we can just `fillna` the other with `fill_value`. Same goes for scalar values because scalars cannot be null (here, when scalar is an input).